src/service_daemon.rs

```diff
@@ -134,18 +138,34 @@ impl ServiceDaemon {
     pub fn new() -> Result<Self> {
         // Use port 0 to allow the system assign a random available port,
         // no need for a pre-defined port number.
-        let signal_sock = UdpSocket::bind("127.0.0.1:0")
+        Self::new_with_signal_addr(SocketAddrV4::new(LOOPBACK_V4, 0).into())
+    }
+
+    /// Creates a new daemon and spawns a thread to run the daemon.
+    ///
+    /// The daemon (re)uses the default mDNS port 5353. To keep it simple, we don't
+    /// ask callers to set the port.
+    /// This one uses an ipv6 local address for the signal mechanism.
+    pub fn new_signal_v6() -> Result<Self> {
```

Comment:
I missed this earlier.  I don't think we need this method. It's okay to just use IPv4 loopback for signaling. 

More importantly, signaling is an implementation detail. We don't want to expose it in public API.

Reply:
I was testing with an IPv6 only host, had an issue and though it was related but it does work using ipv4 local host.
That's a stretch but may there are OSes without an IPv4 stack implementation out there?
Anyway I'll remove the code

Reply:
Thanks! My understanding is that all IP stacks implements at least IPv4. If it turned out there was exceptions, we can always come back and refactor the code to support it.